Library/Homebrew/diagnostic.rb

```diff
@@ -1435,5 +1420,34 @@ def all
         methods.map(&:to_s).grep(/^check_/)
       end
     end # end class Checks
+
+  end
+
+  def inject_dump_stats!(the_module, pattern = /.*/)
```

Comment:
I think I prefer this to have no default for `pattern`. Now that this method is used by multiple commands, it should definitely be moved out of `diagnostic.rb`. Maybe `utils.rb` is a sensible place, probably even better a new file in `utils/`.